### ISSUE TITLE
AutofillGroup on iOS does not fill out the AutofillHints.password field, when the AutofillHints.username field has an existing value

### ISSUE THREAD

**Issue by bhayatus** [flutter/flutter#98619]
Tested on an iPhone 13, with iOS version 15.1.1

Also tested on Android using a Pixel 4a with Android 12, cannot reproduce there.

## Steps to Reproduce

1. Use the following Widget:
```dart
AutofillGroup(
  child: Column(
    children: [
      TextFormField(
        decoration: const InputDecoration(
          label: Text('Email'),
        ),
        keyboardType: TextInputType.emailAddress,
        autofillHints: const [AutofillHints.username],
      ),
      const SizedBox(height: 16),
      TextFormField(
        keyboardType: TextInputType.visiblePassword,
        decoration: const InputDecoration(
          label: Text('Password'),
        ),
        obscureText: true,
        autofillHints: const [AutofillHints.password],
      ),
    ],
  ),
 ),
```

2. Enter an email into the first field.
3. Tap the password field, then proceed with picking a password via any autofill service (Bitwarden for example).

**Expected results:** 
Expected that the password field would be autofilled.

**Actual results:** 
The password field was not autofilled.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: AutofillGroup(
          child: Column(
            children: [
              TextFormField(
                decoration: const InputDecoration(
                  label: Text('Email'),
                ),
                keyboardType: TextInputType.emailAddress,
                autofillHints: const [AutofillHints.username],
              ),
              const SizedBox(height: 16),
              TextFormField(
                keyboardType: TextInputType.visiblePassword,
                decoration: const InputDecoration(
                  label: Text('Password'),
                ),
                obscureText: true,
                autofillHints: const [AutofillHints.password],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
[+18053 ms] Keyboard cannot present view controllers (attempted to present
<UIKeyboardHiddenViewController_ExternalCredential: 0x10650e9f0>)
```

```
flutter analyze
Analyzing src...                                                        
No issues found! (ran in 1.8s)
```

```
flutter doctor -v
[✓] Flutter (Channel stable, 2.10.1, on macOS 12.1 21C52 darwin-arm, locale
    en-CA)
    • Flutter version 2.10.1 at /Users/usaimbhayat/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db747aa133 (7 days ago), 2022-02-09 13:57:35 -0600
    • Engine revision ab46186b24
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from:
      https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK
      components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup
      for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from
      https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup
      for detailed instructions).

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

**Comment by ShabazBadshah** [comment]
+1 I'm also having this exact issue

**Comment by darshankawar** [comment]
@bhayatus 
This seems to be replicable on stable but not on master since this has been fixed recently. https://github.com/flutter/flutter/pull/97846

<img width="346" alt="Screenshot 2022-02-17 at 2 57 23 PM" src="https://user-images.githubusercontent.com/67046386/154446138-c689e7ad-5831-4668-a36d-d794bd0a437c.png">

I suggest you to switch to latest master to see the fix working and probably need to wait till the fix is rolled out on stable.

I verified on `Channel master, 2.11.0-0.0.pre.598`.

With this, I feel safe to close this issue as this is already fixed and available on master. If you disagree, write in comments and I'll reopen it.
